torchaudio/transforms.py

```diff
@@ -284,22 +291,102 @@ class SpectogramToDB(object):
     """
     def __init__(self, stype="power", top_db=None):
         self.stype = stype
-        if top_db is not None and top_db > 0:
-            top_db = -top_db
         self.top_db = top_db
         self.multiplier = 10. if stype == "power" else 20.
+        self.amin = torch.tensor([1e-10])
+        self.ref_value = torch.tensor([1.0])
 
     def __call__(self, spec):
 
-        spec_db = self.multiplier * torch.log10(spec / spec.max())  # power -> dB
+        # numerically stable implementation from librosa
+        # https://librosa.github.io/librosa/_modules/librosa/core/spectrum.html
+        spec_db = self.multiplier * torch.log10(torch.max(self.amin.to(spec.device), spec))
+        spec_db -= self.multiplier * torch.log10(torch.max(self.amin.to(spec.device), self.ref_value.to(spec.device)))
```

Comment:
Since `self.amin` and `self.ref_value` are scalars, it would be much cheaper to treat them as Python floats. The current implementation will result in two host-to-gpu transfers and two CUDA kernel calls to compute what's actually a constant known at construction time.

Reply:
great call.